PowerEditor/src/MISC/PluginsManager/Notepad_plus_msgs.h

```diff
@@ -197,11 +197,31 @@ enum Platform { PF_UNKNOWN, PF_X86, PF_X64, PF_IA64, PF_ARM64 };
 	//void NPPM_TRIGGERTABBARCONTEXTMENU(int view, int index2Activate)
 
 	#define NPPM_GETNPPVERSION (NPPMSG + 50)
-	// int NPPM_GETNPPVERSION(0, 0)
-	// return version
-	// ex : v4.6
-	// HIWORD(version) == 4
-	// LOWORD(version) == 6
+	// int NPPM_GETNPPVERSION(BOOL ADD_ZERO_PADDING, 0)
+	// Get Notepad++ version
+	// HIWORD(returned_value) is major part of version: the 1st number
+	// LOWORD(returned_value) is minor part of version: the 3 last numbers (digits)

```

Comment:
Why not use:
`HIBYTE(HIWORD(returned_value))` for first part of version (major),
`LOBYTE(HIWORD(returned_value))` for second part of version (minor digit 1),
`HIBYTE(LOWORD(returned_value))` for third part of version (minor digit 2),
`LOBYTE(LOWORD(returned_value))` for forth part of version (minor digit 3)?

Reply:
Yes, It's quite clean and clear. But it'll be a new API.

Reply:
Just a sidenote. In my plugins, I have been using this code for getting a properly sortable NPP version:

```C++
float Utils::getNPPVersion() {
   long versionNum{ static_cast<long>(nppMessage(NPPM_GETNPPVERSION, 0, 0)) };

   return std::stof(std::to_wstring(HIWORD(versionNum)) + L"." + std::to_wstring(LOWORD(versionNum)));
}
```

Reply:
@shriprem 
Good demo - that's the way to go!
Do you mind I use your function as document/comment of NPPM_GETNPPVERSION ?

Reply:
Please use it. My pleasure!

Reply:
@donho ,
@shriprem 's approach is very clever but the problem is that plugin developers don't know the Notepad++ versioning convention - at least I didn't know it (maybe I didn't search in the right places).
IMO it will be good to add to `Notepad_plus_msgs.h` as comment to NPPM_GETNPPVERSION the way Notepad++ versions increment (what are the possible values).
For example if Notepad++ supported versions like `8.0.3` @shriprem 's solution would not work (it would be the same as `8.3.0`) and I thought that such versions were completely valid.

Reply:
@pnedev 
> For example if Notepad++ supported versions like 8.0.3 @shriprem 's solution would not work (it would be the same as 8.3.0) and I thought that such versions were completely valid.

Indeed. In this case, it'll be better to use ADD_ZERO_PADDING come with v8.4.1 (ready to go) - so 8.0.3 (LOWORD 030 ie. 30) will be smaller than  v8.3  (LOWORD 300).

I will add a new API in v8.4.2 with the suggestion of @MAPJe71 : 
`HIBYTE(HIWORD(returned_value))` for first part of version (major),
`LOBYTE(HIWORD(returned_value))` for second part of version (minor digit 1),
`HIBYTE(LOWORD(returned_value))` for third part of version (minor digit 2),
`LOBYTE(LOWORD(returned_value))` for forth part of version (minor digit 3).

So there's no more ambiguity.